### ISSUE TITLE
TOC links

### ISSUE THREAD

**Issue by Damix48** [doersino/markdeep-thesis#5]
The links on the TOC don't work neither in the browser nor in the pdf.

Is there any solution?

**Comment by doersino** [comment]
Turns out that there is! Thank you for bringing this to my attention again, I vaguely remember running into this issue myself during initial development, but deciding that it wasn't critical (I had my own thesis to complete and submit, after all).

*First, a bit of background:* Bindery, the library used to turn Markdeep's HTML output into a book-shaped thing, essentially makes a copy of the document during its processing. The original version sticks around, but is hidden – but the anchors contained within remain valid "jump targets" when clicking on an entry of the table of contents. Since they're hidden, the browser effectively jumps to nowhere, so nothing happens.

The fix involves postprocessing the original/leftover document to remove the jump targets. *I think* the *entire* leftover document could be removed, but there might be circumstances (if the document contains, say, JavaScript or embedded videos, perhaps) where it's useful to keep it around.

I'm quite glad I implemented hooks like `hookAfterBindery`, which enables you to **modify your document's `markdeepThesisOptions` as follows to make the links in the table of contents work** both in-browser and in the PDF generated by Chrome:

```
<script>
    markdeepThesisOptions = {
        ...
        hookAfterBindery: () => {
            let hiddenDocument = document.querySelector('body > span.md');  // leftover original document after Bindery has processed it
            hiddenDocument.querySelectorAll("a.target").forEach(e => {  // select all target anchors...
                e.setAttribute("name", "");  // ...and un-target-itfy them
            });
        }
    }
</script>
```

I'll be pushing a fix that does this by default soon, as well.

Again, thank you!
